src/main/java/pacApp/pacSoapConnector/SoapConvertCurrencyConnector.java

```diff
@@ -0,0 +1,53 @@
+package pacApp.pacSoapConnector;
+import org.springframework.ws.client.core.support.WebServiceGatewaySupport;
+import org.springframework.ws.soap.client.core.SoapActionCallback;
+
+import com.consumingwebservice.wsdl.ConvertCurrency;
+import com.consumingwebservice.wsdl.ConvertCurrencyResponse;
+import com.consumingwebservice.wsdl.GetCurrencyCodes;
+import com.consumingwebservice.wsdl.GetCurrencyCodesResponse;
+import com.consumingwebservice.wsdl.ObjectFactory;
+
+import pacApp.pacSoapConnector.context.SoapMarshaller;
+
+public class SoapConvertCurrencyConnector extends WebServiceGatewaySupport {
+	
+	private void setupMarshaller() {
+		getWebServiceTemplate().setMarshaller(new SoapMarshaller().marshaller());
+		getWebServiceTemplate().setUnmarshaller(new SoapMarshaller().marshaller());		
+	}
+	
+	public GetCurrencyCodesResponse getCurrencyCodesResponse() {
+		try {
+			GetCurrencyCodes request = new GetCurrencyCodes();
+			setupMarshaller();
+			GetCurrencyCodesResponse response = (GetCurrencyCodesResponse) getWebServiceTemplate()
+		        .marshalSendAndReceive("http://localhost:50923/Service1.svc/soap", request,
+		            new SoapActionCallback("http://tempuri.org/IService1/GetCurrencyCodes"));
```

Comment:
Tempuri.org ? What is this used for?  the url leads to 404

Reply:
Yes, for this class required!

Reply:
This url is definded in the Soap WSDL